File: 1456.maximum-number-of-vowels-in-a-substring-of-given-length.cs
/*
 * @lc app=leetcode id=1456 lang=csharp
 *
 * [1456] Maximum Number of Vowels in a Substring of Given Length
 */

// @lc code=start
public class Solution 
{
    public int MaxVowels(string s, int k) 
    {
        int current = 0;
        int result = 0;
        int n = s.Length;
        for (int i = 0; i < k; i++)
        {
            if ("aeiou".Contains(s[i]))
            {
                current++;
            }
        }
        result = current;
        for (int i = k; i < n; i++)
        {
            if ("aeiou".Contains(s[i]))
            {
                current++;
            }
            if ("aeiou".Contains(s[i-k]))
            {
                current--;
            }
            result = Math.Max(result, current);
        }
        return result;   
    }
}
// @lc code=end

File: 1456.maximum-number-of-vowels-in-a-substring-of-given-length.py
#
# @lc app=leetcode id=1456 lang=python3
#
# [1456] Maximum Number of Vowels in a Substring of Given Length
#

# @lc code=start
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        current, result, n = 0, 0, len(s)
        for i in range(k):
            if s[i] in "aeiou":
                current += 1
        result = current
        for i in range(k,n,1):
            if s[i] in "aeiou":
                current += 1
            if s[i-k] in "aeiou":
                current -= 1
            result = max(result, current)
        return result
# @lc code=end

